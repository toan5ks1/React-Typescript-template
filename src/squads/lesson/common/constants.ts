import { TypographyBaseProps } from "src/components/Typographys/TypographyBase";
import { TypographyWithValueProps } from "src/components/Typographys/TypographyWithValue";

import { UserGroup } from "manabie-bob/enum_pb";

import { DynamicAutocompleteOptionProps, StudentAttendStatusType } from "./types";

import { convertEnumKeys } from "src/squads/lesson/common/helpers/mapData";
import { UseResourceTranslateReturn } from "src/squads/lesson/hooks/useResourceTranslate";

export enum LabelComponentType {
    TYPOGRAPHY = "TYPOGRAPHY",
    TYPOGRAPHY_WITH_VALUE = "TYPOGRAPHY_WITH_VALUE",
    TYPOGRAPHY_WITH_VALUE_PERCENTAGE = "TYPOGRAPHY_WITH_VALUE_PERCENTAGE",
}

export type LabelComponentTypeRecord = {
    [LabelComponentType.TYPOGRAPHY]: TypographyBaseProps;
    [LabelComponentType.TYPOGRAPHY_WITH_VALUE]: TypographyWithValueProps;
    [LabelComponentType.TYPOGRAPHY_WITH_VALUE_PERCENTAGE]: TypographyWithValueProps;
};

export const generateTranslatedAttendanceStatusOptions = (
    translator: UseResourceTranslateReturn
): DynamicAutocompleteOptionProps[] => {
    const attendanceValuesWithOrder: StudentAttendStatusType[] = [
        "STUDENT_ATTEND_STATUS_ATTEND",
        "STUDENT_ATTEND_STATUS_ABSENT",
        "STUDENT_ATTEND_STATUS_INFORMED_ABSENT",
        "STUDENT_ATTEND_STATUS_LATE",
        "STUDENT_ATTEND_STATUS_INFORMED_LATE",
        "STUDENT_ATTEND_STATUS_LEAVE_EARLY",
    ];

    return attendanceValuesWithOrder.map((attendanceValue) => {
        const attendanceOption: DynamicAutocompleteOptionProps = {
            key: attendanceValue,
            label: translator(`studentAttendanceStatus.${attendanceValue}`),
        };

        return attendanceOption;
    });
};

type LessonReportDetailsKeyPath = `lessonReportDetails.${number}.dynamicFields.${string}`;
export const lessonReportDetailsFormKeyName = (
    fieldIndex: number,
    fieldName: string
): LessonReportDetailsKeyPath => {
    const key = `lessonReportDetails.${fieldIndex}.dynamicFields.${fieldName}`;
    return key as LessonReportDetailsKeyPath;
};

// TODO: talk to BE team to decide on a rule for this name when we have multiple dynamic forms in our system
export const FEATURE_NAME_INDIVIDUAL_LESSON_REPORT_FORM_CONFIG =
    "FEATURE_NAME_INDIVIDUAL_LESSON_REPORT";

export const FEATURE_NAME_GROUP_LESSON_REPORT_FORM_CONFIG = "FEATURE_NAME_GROUP_LESSON_REPORT";

export enum LessonReportSubmittingStatus {
    LESSON_REPORT_SUBMITTING_STATUS_SUBMITTED = "LESSON_REPORT_SUBMITTING_STATUS_SUBMITTED",
    LESSON_REPORT_SUBMITTING_STATUS_SAVED = "LESSON_REPORT_SUBMITTING_STATUS_SAVED",
}

export enum ExtendedUserGroup {
    USER_GROUP_HQ_STAFF = 16,
    USER_GROUP_TEACHER_LEAD = 17,
    USER_GROUP_CENTRE_LEAD = 18,
    USER_GROUP_CENTRE_MANAGER = 19,
    USER_GROUP_CENTRE_STAFF = 20,
}

export const UserGroupUnion = Object.assign({}, UserGroup, ExtendedUserGroup);

export const UserRoles = convertEnumKeys(UserGroupUnion);
