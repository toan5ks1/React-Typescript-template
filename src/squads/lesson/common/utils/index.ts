export * from "./utils";
export * from "./other";
