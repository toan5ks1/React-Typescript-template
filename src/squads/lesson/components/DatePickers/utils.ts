export const isLabelString = (label: React.ReactNode): label is String => {
    return typeof label === "string";
};
