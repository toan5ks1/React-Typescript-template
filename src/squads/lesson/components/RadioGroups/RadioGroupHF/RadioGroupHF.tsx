import { Controller } from "react-hook-form";
import { FormSize } from "src/common/constants/enum";
import { OptionSelectType } from "src/common/constants/types";
import { HookFormControllerOptionProps } from "src/typings/react-hook-form";

import { FormControl, FormControlLabel, FormHelperText, Grid, GridProps } from "@mui/material";
import RadioGroupBase, {
    RadioGroupBaseProps,
} from "src/squads/lesson/components/RadioGroups/RadioGroupBase";
import RadioBase, { RadioBaseProps } from "src/squads/lesson/components/Radios/RadioBase";

interface RadioGroupHFProps
    extends RadioGroupBaseProps,
        Pick<HookFormControllerOptionProps, "rules"> {
    name: string;
    options: OptionSelectType[];
    size?: FormSize;
    sxRadio?: RadioBaseProps["sx"];
    disabled?: boolean;
    directionGird?: GridProps["direction"];
}

const RadioGroupHF = ({ size = "small", sxRadio, ...props }: RadioGroupHFProps) => {
    const {
        defaultValue = "",
        name,
        onChange: handleOnChange,
        options,
        rules,
        disabled,
        directionGird = "row",
        ...rest
    } = props;

    return (
        <FormControl variant="outlined" fullWidth size={size} disabled={disabled}>
            <Controller
                rules={rules}
                defaultValue={defaultValue}
                name={name}
                render={({ field: { onChange, value }, fieldState: { error: fieldError } }) => (
                    <div>
                        <RadioGroupBase
                            onChange={(event, selected) => {
                                onChange(event);
                                handleOnChange && handleOnChange(event, selected);
                            }}
                            row
                            defaultValue={`${defaultValue}`}
                            value={value}
                            {...rest}
                        >
                            <Grid container spacing={1} direction={directionGird}>
                                {options.map((option) => (
                                    <Grid item key={option.id}>
                                        <FormControlLabel
                                            control={
                                                <RadioBase
                                                    sx={sxRadio}
                                                    data-testid={`Radio__${option.id}`}
                                                    size={size}
                                                    color="primary"
                                                />
                                            }
                                            label={option.value}
                                            name={option.value}
                                            value={`${option.id}`}
                                        />
                                    </Grid>
                                ))}
                            </Grid>
                        </RadioGroupBase>
                        {fieldError && (
                            <FormHelperText variant="standard" error={true}>
                                {fieldError.message}
                            </FormHelperText>
                        )}
                    </div>
                )}
            />
        </FormControl>
    );
};

export default RadioGroupHF;
