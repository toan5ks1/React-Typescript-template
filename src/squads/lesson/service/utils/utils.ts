import { ArrayElement } from "src/common/constants/types";
import { Media } from "src/squads/lesson/common/types";
import { ArrayOrSetValue } from "src/squads/lesson/service/bob/lesson-reports-service/types";
import { NsLesson_Bob_UploadService } from "src/squads/lesson/service/bob/upload-service/types";

import { Paging } from "manabuf/common/v1/requests_pb";

export function createEmptyResponse<T = any>(resp: T): Promise<T> {
    return Promise.resolve(resp);
}

export function getEmptyResponse() {
    return createEmptyResponse({
        data: {
            id: null,
        },
    });
}

export function getSearchString(text: string | null | undefined) {
    if (!text) return undefined;
    return `%${text}%`;
}

/**
 * Convert to "Paging" proto with only limit and offset
 *
 * @param paging paging object {limit: number; offsetInteger: number; offsetString: string; ...}
 * @returns proto class "Paging"
 */
export const toBasicPagingProto = (paging: Paging.AsObject) => {
    const result = new Paging();

    result.setLimit(paging.limit);

    // We only set the offset by one of 2 things, string or integer
    if (Boolean(paging.offsetString)) result.setOffsetString(paging.offsetString);
    else result.setOffsetInteger(paging.offsetInteger);

    return result;
};

export const toNonUndefinedArrayValueList = <T extends ArrayElement<ArrayOrSetValue>>(
    valuesList: T[] | undefined
): T[] => {
    if (valuesList && Array.isArray(valuesList)) return valuesList.map((value) => value);

    return [];
};

export const getFileExtension = (filename: string) => {
    return filename.split(".").pop() || "";
};

export const getFileNameWithoutExtension = (filename: string) => {
    return filename.replace(`.${getFileExtension(filename)}`, "");
};

export const isMedia = (media: NsLesson_Bob_UploadService.FilterUploadFiles): media is Media => {
    return (media as Media).media_id !== undefined;
};

export const filterUploadFiles = (files: NsLesson_Bob_UploadService.FilterUploadFiles[]) => {
    const alreadyUploadedMediaIds: string[] = [];
    const filesNeedToBeUploaded: File[] = [];
    for (let i = 0; i < files.length; i++) {
        const current = files[i];
        if (isMedia(current)) {
            alreadyUploadedMediaIds.push(current.media_id);
            continue;
        }
        if (current instanceof File) {
            filesNeedToBeUploaded.push(current);
        }
    }
    return { alreadyUploadedMediaIds, filesNeedToBeUploaded };
};
