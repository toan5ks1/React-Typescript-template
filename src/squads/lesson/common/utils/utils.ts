import { DateTime } from "luxon";
import { choiceDayOfWeek } from "src/common/constants/choices";
import { FormatDateOptions } from "src/common/constants/enum";
import { ArrayElement, DateCommon } from "src/common/constants/types";
import { arrayHasItem } from "src/common/utils/other";
import {
    DynamicAutocompleteOptionProps,
    DynamicFormFieldValue,
    TimeAutocompleteOption,
} from "src/squads/lesson/common/types";
import { Lesson_LessonByLessonIdForLessonManagementV3Query } from "src/squads/lesson/service/bob/bob-types";

import { UseTranslateReturn } from "src/squads/lesson/hooks/useTranslate";

export const isDynamicAutocompleteOptionProps = (
    value?: DynamicFormFieldValue
): value is DynamicAutocompleteOptionProps => {
    return value && value["key"] && value["label"] && Object.keys(value).length === 2;
};

export const convertToNumber = (value?: DynamicFormFieldValue): number => {
    if (Number.isNaN(Number(value))) {
        return 0;
    }

    return Number(value);
};

export const convertToBoolean = (value?: DynamicFormFieldValue): boolean => {
    return String(value).toLowerCase() === "true";
};

export const convertToString = (value?: DynamicFormFieldValue): string => {
    return typeof value === "string" ? value : "";
};

export function convertToArray<T>(array: T): T[] {
    return Array.isArray(array) && arrayHasItem(array) ? array : [];
}

export const getTeacherName = (
    teachersList: ArrayElement<
        Lesson_LessonByLessonIdForLessonManagementV3Query["lessons"]
    >["lessons_teachers"]
) => {
    if (!arrayHasItem(teachersList)) return emptyValue;

    const nonUndefinedNames = teachersList.reduce((result: string[], currentTeacher) => {
        const name = currentTeacher.teacher.users?.name;
        if (name) result.push(name);
        return result;
    }, []);
    const teacherNames = nonUndefinedNames.join(", ");
    return teacherNames;
};

export const getDayOfWeekName = (date: Date, translator: UseTranslateReturn) => {
    const toDay = new Date(date).getDay();
    return translator(choiceDayOfWeek[toDay].name);
};

export const getDurationBetweenTwoDates = (startTime: DateTime, endTime: DateTime) => {
    const startDate = DateTime.fromISO(`${startTime}`);
    const endDate = DateTime.fromISO(`${endTime}`);

    const duration = endDate.diff(startDate, ["hours", "minutes"]);

    return duration.hours * 60 + duration.minutes;
};

export const formatDate = (date: DateCommon["date"], formatOptions: FormatDateOptions) => {
    const dt = new Date(date);
    return DateTime.fromJSDate(dt).toFormat(formatOptions);
};

export const createTimeList = (
    baseDate?: TimeAutocompleteOption["value"]
): TimeAutocompleteOption[] => {
    const timeList: TimeAutocompleteOption[] = [];
    const date = baseDate ? new Date(baseDate) : new Date();

    // Start from 00:00 AM
    date.setHours(0, 0, 0, 0);

    // Iterate to generate time list for a day (1440 minutes)
    for (let i = 0; i < 1440; i++) {
        const hour = String(date.getHours()).padStart(2, "0");
        const minute = String(date.getMinutes()).padStart(2, "0");

        const today = new Date(
            date.getFullYear(),
            date.getMonth(),
            date.getDate(),
            Number(hour),
            Number(minute)
        );

        timeList.push({ label: formatDate(today, "HH:mm"), value: today });
        date.setMinutes(date.getMinutes() + 1);
    }

    return timeList;
};

export const timeOptions = createTimeList();

export const emptyValue = "--";
