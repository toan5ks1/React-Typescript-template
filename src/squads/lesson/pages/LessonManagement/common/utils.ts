import { convertEnumKeys } from "src/common/constants/helper";
import { arrayHasItem } from "src/common/utils/other";
import { generateTranslatedAttendanceStatusOptions } from "src/squads/lesson/common/constants";
import {
    DynamicAutocompleteOptionProps,
    DynamicFieldProps,
    DynamicSection,
    ValueTypeNull,
} from "src/squads/lesson/common/types";

import { ValueType } from "manabuf/bob/v1/lessons_pb";
import { LessonSchedulingStatus } from "manabuf/common/v1/enums_pb";

import { UseResourceTranslateReturn } from "src/squads/lesson/hooks/useResourceTranslate";
import {
    DynamicFieldComponent,
    LessonReportSubmittingStatusKeys,
    LessonReportSubmittingStatusType,
    LessonStatusType,
    StaticFieldComponent,
} from "src/squads/lesson/pages/LessonManagement/common/types";

/**
 * Get value type of a dynamic field
 * @param fieldId id of a dynamic field which is contained by sections of formSections
 * @param formSections sections of "partner form config" - Example: "mockDataConfig" from {@link src/squads/lesson/test-utils/lesson-report.ts}
 *
 * @returns value type in manabuf {@link ValueType from "manabuf/bob/v1/lessons_pb"} (mean not include VALUE_TYPE_NULL)
 * or undefined if @param formSections not contain dynamic field has @param fieldId
 */
export const getNonNullableValueType = (
    fieldId: DynamicFieldProps["field_id"],
    formSections: DynamicSection[]
): ValueType | undefined => {
    if (!formSections || !arrayHasItem(formSections)) return undefined;

    const valueTypeNull: keyof typeof ValueTypeNull = "VALUE_TYPE_NULL";
    let valueType: ValueType | undefined = undefined;

    const allSectionFields = formSections.flatMap((section) => section.fields);

    const targetField = allSectionFields.find((field) => field.field_id === fieldId);

    const isValuableType = targetField && targetField.value_type !== valueTypeNull;

    if (isValuableType && targetField) {
        valueType = ValueType[targetField.value_type];
    }

    return valueType;
};

/**
 * Get component type of dynamic field
 * @param fieldId id of a dynamic field which is contained by sections of formSections
 * @param formSections sections of "partner form config" - Example: "mockDataConfig" from {@link src/squads/lesson/test-utils/lesson-report.ts}
 *
 * @returns component type of {@link DynamicFieldsComponentType/LessonReportIndividualDynamicComponentType from "src/squads/lesson/common/types"}
 * or undefined if @param formSections not contain dynamic field has @param fieldId
 */
export const getComponentType = (
    fieldId: DynamicFieldProps["field_id"],
    formSections: DynamicSection[]
) => {
    if (!formSections || !arrayHasItem(formSections)) return undefined;

    let componentType: DynamicFieldProps["component_config"]["type"] | undefined = undefined;

    const allSectionFields = formSections.flatMap((section) => section.fields);

    const targetField = allSectionFields.find((field) => field.field_id === fieldId);

    if (targetField) {
        componentType = targetField.component_config.type;
    }

    return componentType;
};

/**
 * Get attendance status option for autocomplete
 * @param fieldId string value from lesson report detail
 * @param translator translate function
 *
 * @returns option for autocomplete {@link DynamicAutocompleteOptionProps src/squads/lesson/common/types}
 */
export const getAttendanceStatusOption = (
    fieldValue: DynamicAutocompleteOptionProps["key"],
    translator: UseResourceTranslateReturn
) => {
    const attendanceStatus = generateTranslatedAttendanceStatusOptions(translator);
    return attendanceStatus.find((status) => status.key === fieldValue);
};

export const isDynamicFieldComponent = (
    type: DynamicFieldComponent | StaticFieldComponent
): type is DynamicFieldComponent => {
    return Object.keys(convertEnumKeys(DynamicFieldComponent)).includes(type);
};

type AllowSaveDraftLessonType = Exclude<
    LessonStatusType,
    "LESSON_SCHEDULING_STATUS_COMPLETED" | "LESSON_SCHEDULING_STATUS_CANCELED"
>;

export const isAllowSaveDraftLesson = (
    lessonStatus: string
): lessonStatus is AllowSaveDraftLessonType => {
    if (!lessonStatus) return true;

    return [
        LessonSchedulingStatus.LESSON_SCHEDULING_STATUS_DRAFT,
        LessonSchedulingStatus.LESSON_SCHEDULING_STATUS_PUBLISHED,
    ].includes(LessonSchedulingStatus[lessonStatus]);
};

export const isLessonReportStatusType = (
    lessonReportStatus: string
): lessonReportStatus is LessonReportSubmittingStatusType => {
    if (!lessonReportStatus) return false;
    return Object.keys(LessonReportSubmittingStatusKeys).includes(lessonReportStatus);
};
