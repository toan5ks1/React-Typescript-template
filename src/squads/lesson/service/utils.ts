import { toArr } from "src/common/utils/other";
import { RaSort } from "src/squads/lesson/typings/react-admin";

import { Paging } from "manabuf/common/v1/requests_pb";

import { GqlSort } from "./service-types";

import { Order_By } from "src/squads/lesson/__generated__/bob/root-types";

export function createEmptyResponse<T = any>(resp: T): Promise<T> {
    return Promise.resolve(resp);
}

export function getEmptyResponse() {
    return createEmptyResponse({
        data: {
            id: null,
        },
    });
}

export function toGqlQuerySorts(sorts?: RaSort | RaSort[] | GqlSort): GqlSort {
    if (!sorts) return { created_at: Order_By.Desc };
    const raSorts = toArr(sorts) as RaSort[];

    if (!raSorts[0]?.order || !raSorts[0]?.field) {
        return sorts as GqlSort;
    }

    let result: GqlSort = {};
    raSorts.map((e) => {
        result[e.field] = e.order.toLowerCase() as unknown as Order_By;
    });

    return result;
}

/**
 * Convert to "Paging" proto with only limit and offset
 *
 * @param paging paging object {limit: number; offsetInteger: number; offsetString: string; ...}
 * @returns proto class "Paging"
 */
export const toBasicPagingProto = (paging: Paging.AsObject) => {
    const result = new Paging();

    result.setLimit(paging.limit);

    // We only set the offset by one of 2 things, string or integer
    if (Boolean(paging.offsetString)) result.setOffsetString(paging.offsetString);
    else result.setOffsetInteger(paging.offsetInteger);

    return result;
};
