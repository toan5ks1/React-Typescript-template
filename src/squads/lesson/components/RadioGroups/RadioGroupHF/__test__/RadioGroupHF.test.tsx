import { ERPModules } from "src/common/constants/enum";
import { convertToChoices } from "src/common/utils/choice";

import RadioGroupHF from "../RadioGroupHF";

import { cleanup, fireEvent, render, RenderResult, screen } from "@testing-library/react";
import TestHookFormProvider from "src/squads/lesson/test-utils/TestHookFormProvider";

export const ScheduleKeys = {
    OPENING_STATUS: "openingStatus",
    REPEAT_TYPE: "repeatType",
    SCHEDULE_SEARCH_TYPE: "scheduleSearchType",
    DAY_CONDITION_TYPE: "dayConditionType",
};

export enum DayConditionType {
    ALL_DAY = 0,
    SPECIFIC_TIME = 1,
}

export const choiceDayConditionType = convertToChoices(
    DayConditionType,
    ScheduleKeys.DAY_CONDITION_TYPE,
    ERPModules.SCHEDULE
);

const defaultProps = {
    name: "name",
    label: "Name",
    options: choiceDayConditionType,
    "data-testid": "RadioGroupHF__container",
    onChange: jest.fn(),
    rules: {
        required: { value: true, message: "Name is required" },
    },
};
const HookFormComponent = (restProps: any) => {
    return (
        <TestHookFormProvider>
            <RadioGroupHF {...defaultProps} {...restProps} />
        </TestHookFormProvider>
    );
};

describe("<RadioGroupHF />", () => {
    let wrapper: RenderResult;

    afterEach(cleanup);

    beforeEach(() => {
        wrapper = render(<HookFormComponent />);
    });

    it("should exist label", () => {
        expect(
            screen.getByText(
                `resources.schedule.choices.dayConditionType.${DayConditionType.ALL_DAY}`
            )
        ).toBeInTheDocument();
        expect(
            screen.getByText(
                `resources.schedule.choices.dayConditionType.${DayConditionType.SPECIFIC_TIME}`
            )
        ).toBeInTheDocument();
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });

    it("should exist error", async () => {
        expect(wrapper.queryByTestId("RadioGroupHF__container")).toBeInTheDocument();
        expect(wrapper.queryByTestId("TextFieldHF__submit")).toBeInTheDocument();

        fireEvent.click(wrapper.getByTestId("TextFieldHF__submit"));

        expect(await wrapper.findByText("Name is required")).toBeInTheDocument();
    });

    it("should show RadioGroupHF__container with type Specific Time", () => {
        const specificRadioButton = wrapper.getByTestId(`Radio__${DayConditionType.SPECIFIC_TIME}`);
        fireEvent.click(specificRadioButton as HTMLInputElement);

        expect(
            wrapper.queryByTestId(`Radio__${DayConditionType.SPECIFIC_TIME}`)
        ).toBeInTheDocument();
    });
});
