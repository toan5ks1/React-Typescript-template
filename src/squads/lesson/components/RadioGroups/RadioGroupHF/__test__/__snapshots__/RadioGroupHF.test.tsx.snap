// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroupHF /> should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: -8px;
  width: calc(100% + 8px);
  margin-left: -8px;
}

.emotion-2>.MuiGrid-item {
  padding-top: 8px;
}

.emotion-2>.MuiGrid-item {
  padding-left: 8px;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  margin-left: -11px;
  margin-right: 16px;
}

.emotion-4.Mui-disabled {
  cursor: default;
}

.emotion-4 .MuiFormControlLabel-label.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  padding: 9px;
  border-radius: 50%;
  color: rgba(0, 0, 0, 0.6);
}

.emotion-5::-moz-focus-inner {
  border-style: none;
}

.emotion-5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-5 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-5:hover {
  background-color: rgba(25, 118, 210, 0.04);
}

@media (hover: none) {
  .emotion-5:hover {
    background-color: transparent;
  }
}

.emotion-5.Mui-checked {
  color: #1976d2;
}

.emotion-5.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-6 {
  cursor: inherit;
  position: absolute;
  opacity: 0;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.emotion-7 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-8 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.25rem;
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-9 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.25rem;
  left: 0;
  position: absolute;
  -webkit-transform: scale(0);
  -moz-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: -webkit-transform 150ms cubic-bezier(0.4, 0, 1, 1) 0ms;
  transition: transform 150ms cubic-bezier(0.4, 0, 1, 1) 0ms;
}

.emotion-10 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-11 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

<div>
  <form
    novalidate=""
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth emotion-0"
    >
      <div>
        <div
          class="MuiFormGroup-root MuiFormGroup-row emotion-1"
          data-testid="RadioGroupHF__container"
          label="Name"
          role="radiogroup"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 emotion-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item emotion-3"
            >
              <label
                class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd emotion-4"
              >
                <span
                  class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root emotion-5"
                  data-testid="Radio__0"
                >
                  <input
                    class="PrivateSwitchBase-input emotion-6"
                    name="resources.schedule.choices.dayConditionType.0"
                    type="radio"
                    value="0"
                  />
                  <span
                    class="emotion-7"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-8"
                      data-testid="RadioButtonUncheckedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-9"
                      data-testid="RadioButtonCheckedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root emotion-10"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label emotion-11"
                >
                  resources.schedule.choices.dayConditionType.0
                </span>
              </label>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item emotion-3"
            >
              <label
                class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd emotion-4"
              >
                <span
                  class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root emotion-5"
                  data-testid="Radio__1"
                >
                  <input
                    class="PrivateSwitchBase-input emotion-6"
                    name="resources.schedule.choices.dayConditionType.1"
                    type="radio"
                    value="1"
                  />
                  <span
                    class="emotion-7"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-8"
                      data-testid="RadioButtonUncheckedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-9"
                      data-testid="RadioButtonCheckedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root emotion-10"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label emotion-11"
                >
                  resources.schedule.choices.dayConditionType.1
                </span>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
    <input
      data-testid="TextFieldHF__submit"
      type="submit"
    />
  </form>
</div>
`;
